Update 2019-08-23-guide_install-css-assets.md (#162)
momentum-mod/docs: _posts/guides/2019-08-23-guide_install-css-assets.md

  - cs:s
---

> This Guide covers how to install CS:S assets from SteamCMD.

Reviewer: ```suggestion
This guide covers how to install CS:S assets using SteamCMD.
```

## Prerequisites 
- An internet connection.

> - some free space on your pc. (~20GB)

Reviewer: ```suggestion
- Some free space on your PC. (~20GB)
```
Is it really that much? When doing it myself it was under 4GB.

Author: Indeed, it's only 4GB

## Getting SteamCMD

- Make a folder for SteamCMD.
- Inside of it download [***SteamCMD***](https://steamcdn-a.akamaihd.net/client/installer/steamcmd.zip), and unzip it. 

> - Start steamcmd.exe and wait until you got 'Steam>' waiting for your input.

Reviewer: This is assuming Windows. Can just link to Valve's guide for Linux/Mac: https://developer.valvesoftware.com/wiki/SteamCMD

## Getting CSS files

> inside SteamCMD, type or paste each of the following line in the order :

Reviewer: ```suggestion
Inside SteamCMD, type or paste each of the following line in the order:
```

- `login anonymous`
- `force_install_dir ./css`
- `app_update 232330 validate`

Now for unpacking it do :
- Hold Ctrl+Shift then right click on the 'css' folder and select "Open command window here" or "Open PowerShell here".
- Inside of it type or paste `.\bin\vpk.exe .\cstrike\cstrike_pak_dir.vpk`. (wait for it to do his stuff)
- Now everything is extracted and should be in a folder called 'cstrike_pak_dir' in 'css/cstrike'.

> - from 'css/cstrike' copy the 'materials', 'models' and 'sound' folder and paste it to your Momentum folder.

Reviewer: A better way would be to move the VPKs from here into the momentum folder & editing `gameinfo.txt` to load them. VPKs in general are much more efficient for the engine to load than loose files.

> 
> ## Test and cleaning
> now everything should be good.
> - Start the game and load a map with css assets,if everything looks normal you are done. 
> - you can now delete the whole SteamCMD folder.

Reviewer: ```suggestion
```
I believe this section is redundant.

  - cs:s
---

This guide covers how to install CS:S assets using SteamCMD.
## Prerequisites 
- An internet connection.
- Some free space on your PC. (~4GB)

## Getting SteamCMD
For Linux/Mac refer to Valve's guide:

> https://developer.valvesoftware.com/wiki/SteamCMD

Reviewer: Should be a click-able link.

### Windows:
- Make a folder for SteamCMD.
- Inside of it download [***SteamCMD***](https://steamcdn-a.akamaihd.net/client/installer/steamcmd.zip), and unzip it. 

> - Start steamcmd.exe and wait until you got 'Steam>' waiting for your input.

Reviewer: ```suggestion
- Start steamcmd.exe and wait until you got `Steam>` waiting for your input.
```

## Getting CS:S files

Inside SteamCMD, type or paste each of the following line in the order:
- `login anonymous`
- `force_install_dir ./css`
- `app_update 232330 validate`

## Installing CS:S files

> - Inside your css folder go into the folder `cstrike` and copy/cut all `.vpk` files starting with the name `cstrike_pak_`.

Reviewer: ```suggestion
- Inside your SteamCMD folder go into the folder `css/cstrike` and copy/cut all `.vpk` files starting with the name `cstrike_pak_`.
```

  - cs:s
---

This guide covers how to install CS:S assets using SteamCMD.
## Prerequisites 
- An internet connection.
- Some free space on your PC. (~4GB)

## Getting SteamCMD
For Linux/Mac refer to Valve's guide:
https://developer.valvesoftware.com/wiki/SteamCMD
### Windows:
- Make a folder for SteamCMD.
- Inside of it download [***SteamCMD***](https://steamcdn-a.akamaihd.net/client/installer/steamcmd.zip), and unzip it. 
- Start steamcmd.exe and wait until you got 'Steam>' waiting for your input.

## Getting CSS files

Inside SteamCMD, type or paste each of the following line in the order:
- `login anonymous`
- `force_install_dir ./css`
- `app_update 232330 validate`

## Installing CSS files

> - Inside your css folder move the folder `cstrike` at the root of the momentum folder.

Reviewer: There are a lot of other resources in the `cstrike` folder that we don't need here. I would just dump the needed VPKs into the momentum folder and add `game momentum/cstrike_pak_dir.vpk` to the gameinfo.
![image](https://user-images.githubusercontent.com/9014762/104833982-74329780-5851-11eb-89ba-b6455cc9cf3f.png)

Just trying to get around including more files than we need to.

Reviewer 2: I would chime in here and say that the best place for it would most likely be in a `momentum/custom/` folder, like `momentum/custom/cstrikesource/`

Author: yea you'r right about `cstrike`.

But I think those files need to be placed insinde there own folder not just dump into the momentum one.
maybe `momentum/dependencies`  ? 

Gocnak `momentum/custom/cstrikesource/` seems nice too.

- Inside the momentum folder open `momentum\gameinfo.txt`:
  - Find:
  ```
  game                "C:\Program Files (x86)\Steam\steamapps\common\Counter-Strike Source\cstrike\cstrike_pak.vpk"
  game                "C:\Program Files (x86)\Steam\steamapps\common\Counter-Strike Source\cstrike\download"
  game                "C:\Program Files (x86)\Steam\steamapps\common\Counter-Strike Source\cstrike"

>   ```

Reviewer: Most players are not going to have this in their gameinfo, I'm guessing you did originally to get it to manually mount CS:S stuff.

Author: Those lines are present by default in the gameinfo.txt from 0.8.7-Public-Win.

   - And replace it by:
  ```
  game				|all_source_engine_paths|cstrike/cstrike_pak.vpk
  game				|all_source_engine_paths|cstrike/download

>   game				|all_source_engine_paths|cstrike

Reviewer: I believe you only want the VPK here.
